bionic/cache.py

```diff
@@ -271,7 +271,10 @@ def _get_nearest_entry_with_artifact(self):
         return None
 
     def _get_local_entry(self):
-        if self._stored_local_entry is None:
+        if (
```

Comment:
Yeah, this one is a little gross but it was the quickest way to fix it. 

Any lookup done in the subprocess isn't communicated back so this really only happens when someone does a lookup in main process before the entity is computed and the entity is computed in subprocess. This only happens during assisted versioning when we try to load provenance for checking any versioning issues. When the cache doesn't exist, this entry is set to a not matched entry. When computed in subprocess, the updated entry set isn't populated back (which we do on reregister results by clearing stored entries) and any time someone does a get in main process, it returns a not matched entry with no result.

Some other ways we can consider are:
- peek provenance instead of loading provenance during versioning check.
- force reload entries inside the main process as soon as they are computed in subprocess. 

But to answer your question directly, this would increase the number of lookup at most once for an entity computed in subprocess but called in main process, and only when we are in assisted versioning mode.

Reply:
I'm more concerned about the number of times this will happen during normal (non-parallel) operation. _get_local_entry gets called several times for each entity, so if the artifact is not there it will now attempt to look it up each time. This is not a huge deal but it does add a lot of little filesystem operations.

I like your "force reload" idea -- maybe by adding a method that flushes the local entries on an accessor. In general, it would be nice to keep this new parallel code as isolated as possible until it's ready for prime time.